prymer/variant.py

```diff
@@ -0,0 +1,259 @@
+import logging
+from dataclasses import dataclass
+from pathlib import Path
+from types import TracebackType
+from typing import ContextManager
+from typing import Optional
+
+import pysam
+from pysam import VariantFile
+from pysam import VariantRecord
+
+from prymer.model import Span
+from prymer.model import Strand
+
+_logger = logging.getLogger(__name__)
+
+
+@dataclass(slots=True, frozen=True, init=True)
+class SimpleVariant:
+    """Represents a variant from a given genomic range.
+
+    Each variant has a single reference and alt allele.  Start and end positions are both 1-based
+    inclusive.  Unlike VCF, empty alleles are allowed and positions are given as follows:
+
+    - SNPs and MNPS will have start/end set to the first and last substituted bases
+    - Deletions will have start/end set to the first and last _deleted_ base
+    - Insertions will have start/end set to the bases between which the insertion occurs
```

Comment:
I get what you mean by _between_, but it just doesn't track to say "between an insertion" for me 
```suggestion
    - Insertions will have start/end set to the bases before and after which the insertion occurs respectiely
```

Reply:
I worded slightly differently but close to what you suggested.